### PR TITLE
Add variable constructor

### PR DESCRIPTION
**Pull request by emanguy**
* Add variable model constructor and unit tests

**TO DO STILL**
 - [x] Merge changes from #21 once it passes

### CODE REVIEW

File: src/js/AMPSetup/AMPSetup.js [hunk 1]
```diff
@@ -78,3 +81,88 @@ AMPHexConsumer.prototype.consumeNibbles = function(nibbles)
 	return this.consumedHex;
 }
 
+/**
+<<<<<<< HEAD
```
Comment (gti28) [inline src/js/AMPSetup/AMPSetup.js]: fix merge errro

File: src/js/AMPSetup/AMPSetup.js [hunk 2]
```diff
@@ -78,3 +81,88 @@ AMPHexConsumer.prototype.consumeNibbles = function(nibbles)
...
+	17: "Timestamp",
+	18: "String",
+	19: "Blob",
+	20: "Managed Identifier",
+	21: "MID Collection",
+	22: "Expression",
+	23: "Data Collection",
+	24: "Typed Data Collection",
+	25: "Table"
+=======
```
Comment (gti28) [inline src/js/AMPSetup/AMPSetup.js]: fix merge error

File: src/js/AMPSetup/AMPSetup.js [hunk 3]
```diff
@@ -78,3 +81,88 @@ AMPHexConsumer.prototype.consumeNibbles = function(nibbles)
...
+	17: [ "buildTimestamp" ],
+	18: [ "buildString" ],
+	19: [ "buildBlob" ],
+	20: [ "buildManagedIdentifier" ],
+	21: [ "buildMIDCollection" ],
+	22: [ "buildExpression" ],
+	23: [ "buildDataCollection", false ],
+	24: [ "buildDataCollection", true ],
+	25: [ "buildTable" ]
+>>>>>>> tmaher1/dataCollection
```
Comment (gti28) [inline src/js/AMPSetup/AMPSetup.js]: fix merge error

File: src/js/model/variable/variable.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+	{
+		err.nibblesConsumed += consumedNibbles;
+		err.message = "Unknown type enumeration: " + tempValue.value;
+		throw err;
+	}
+
+	// Build the expression necessary for initializing the variable
+	try
+	{
+		tempValue = AMPParse.buildMidCollection(hexadecimal);
```
Comment (gti28) [inline src/js/model/variable/variable.js]: should this be buildExpression instead of MidCollection?

**Review by gti28 — DISMISSED** [review]

File: src/js/model/variable/variable.js (hunk 4 above)
Comment (emanguy) [reply src/js/model/variable/variable.js]: Changed.

**Pull request merged**